controller_manager/test/test_load_controller.cpp

```diff
@@ -49,198 +66,130 @@ TEST_F(TestLoadController, load_controller_failed_init)
-  // load and start controller with name1
-  {
-    auto switch_future = std::async(
-      std::launch::async, &controller_manager::ControllerManager::switch_controller, cm_,
-      strvec{controller_name1}, strvec{}, STRICT, true, rclcpp::Duration(0, 0));
-
-    ASSERT_EQ(std::future_status::timeout, switch_future.wait_for(std::chrono::milliseconds(100)))
-      << "switch_controller should be blocking until next update cycle";
-    ControllerManagerRunner cm_runner(this);
-    EXPECT_EQ(controller_interface::return_type::OK, switch_future.get());
-  }
-  // Stop controller
-  RCLCPP_INFO(cm_->get_logger(), "Stopping started controller");
-  auto switch_future = std::async(
-    std::launch::async, &controller_manager::ControllerManager::switch_controller, cm_, strvec{},
-    strvec{controller_name1}, STRICT, true, rclcpp::Duration(0, 0));
-
-  ASSERT_EQ(std::future_status::timeout, switch_future.wait_for(std::chrono::milliseconds(100)))
-    << "switch_controller should be blocking until next update cycle";
-  ControllerManagerRunner cm_runner(this);
-  EXPECT_EQ(controller_interface::return_type::OK, switch_future.get());
-
-  ASSERT_EQ(lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE, controller_if->get_state().id());
-}
-
-TEST_F(TestLoadController, can_not_start_finalized_controller)
+TEST_F(TestLoadedController, can_not_start_finalized_controller)
 {
-  // load and start controller with name1
-  auto controller_if = cm_->load_controller(controller_name1, TEST_CONTROLLER_CLASS_NAME);
-  ASSERT_NE(controller_if, nullptr);
-
   ASSERT_EQ(
     lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED, controller_if->get_state().id());
 
   // Shutdown controller on purpose for testing
   ASSERT_EQ(controller_if->shutdown().id(), lifecycle_msgs::msg::State::PRIMARY_STATE_FINALIZED);
 
   //  Start controller
-  strvec start_controllers = {controller_name1};
-  strvec stop_controllers = {};
-  auto switch_future = std::async(
-    std::launch::async, &controller_manager::ControllerManager::switch_controller, cm_,
-    start_controllers, stop_controllers, STRICT, true, rclcpp::Duration(0, 0));
-
-  ASSERT_EQ(std::future_status::ready, switch_future.wait_for(std::chrono::milliseconds(100)))
-    << "switch_controller should be blocking until next update cycle";
-  ControllerManagerRunner cm_runner(this);
-  EXPECT_EQ(controller_interface::return_type::ERROR, switch_future.get());
+  start_test_controller(std::future_status::ready, controller_interface::return_type::ERROR);
 
   // Can not configure unconfigured controller
   EXPECT_EQ(cm_->configure_controller(controller_name1), controller_interface::return_type::ERROR);
   ASSERT_EQ(lifecycle_msgs::msg::State::PRIMARY_STATE_FINALIZED, controller_if->get_state().id());
 }
 
-TEST_F(TestLoadController, inactive_controller_cannot_be_cleaned_up)
+TEST_F(TestLoadedController, inactive_controller_cannot_be_cleaned_up)
```

Comment:
You did everything right, but I think that this test is conceptually wrong. We should be able to clean-up inactive controller to get into `unconfigured state`. Should we open an follow-up issue for this? @bmagyar

Reply:
Where can I read more on those concepts? 
In any case, I think this is out of the scope of the PR, no?